ultravox/tools/ds_tool.py

```diff
@@ -0,0 +1,172 @@
+import dataclasses
+import os
+from typing import Any, Dict, Optional, Union
+
+import datasets
+import openai
+import simple_parsing
+
+from ultravox.tools import tts
+
+CHAT_CLIENT = openai.Client()
+TTS_CLIENT = tts.AzureTts()
+
+DEFAULT_TEXTGEN_TEMPLATE = """Passage: {passage}
+
+Question: {question}
+
+Answer: {answer}
+
+Provide a short explanation to the question given the passage that entails the answer."""
+
+
+@dataclasses.dataclass(frozen=True)
+class TtsTask:
+    column_name: str = simple_parsing.field(default="question", alias="-c")
+    audio_column_name: Optional[str] = simple_parsing.field(default=None, alias="-a")
+    voice: Optional[str] = simple_parsing.field(default=None, alias="-V")
+    sample_rate: int = simple_parsing.field(default=16000, alias="-r")
+
+
+@dataclasses.dataclass(frozen=True)
+class TextGenerationTask:
+    language_model: str = simple_parsing.field(default="gpt-4o", alias="-m")
+    new_column_name: str = simple_parsing.field(default="explanation", alias="-c")
+    template: str = DEFAULT_TEXTGEN_TEMPLATE
+
+
+# This script is used to either generate audio samples from text using a TTS model, or to generate text samples using a text generation model.
+# Ex: just ds_tool -T tts -d google/boolq -u fixie-ai/boolq-audio -c question -a audio
+# Ex: just ds_tool -T textgen -d fixie-ai/boolq-audio -u fixie-ai/boolq-audio -c explanation
+# Ex: just ds_tool -T textgen -d ylacombe/expresso -u fixie-ai/expresso -c continuation \
+#         --template "\"Continue the following sentence in a way that reflects a ‘{style}’ tone in a coherent style:\n{text}\""
+@dataclasses.dataclass
+class DatasetToolArgs:
+    dataset_name: str = simple_parsing.field(alias="-d")
+    dataset_subset: str = simple_parsing.field(default="default", alias="-S")
+    dataset_split: Optional[str] = simple_parsing.field(default=None, alias="-s")
+
+    num_samples: Optional[int] = simple_parsing.field(default=None, alias="-n")
+    num_workers: int = simple_parsing.field(default=16, alias="-w")
+
+    upload_name: Optional[str] = simple_parsing.field(default=None, alias="-u")
+    upload_branch: Optional[str] = simple_parsing.field(default="main", alias="-b")
+    num_shards: Optional[int] = simple_parsing.field(default=None, alias="-N")
+
+    token: Optional[str] = simple_parsing.field(default=None, alias="-t")
+
+    task: Union[TtsTask, TextGenerationTask] = simple_parsing.subgroups(
+        {"tts": TtsTask, "textgen": TextGenerationTask},  # type: ignore
+        default_factory=TtsTask,
+        alias="-T",
```

Comment:
Ideally this wouldn't need an argument name, we could do this as a positional parameter (e.g., `ds_tool tts`) Doesn't need to be solved in this PR though if it's tricky.

Reply:
If we keep the param name though I'd make the alias "-t" for `task` and "-T" for `template` and remove the alias for `token`.